internal/apps/dop/component-protocol/components/issue-manage/batchOperationTipModal/render.go

```diff
@@ -0,0 +1,175 @@
+		return &BatchOperationTipModal{}
+	})
+}
+func (bot *BatchOperationTipModal) Render(ctx context.Context, c *cptype.Component, scenario cptype.Scenario, event cptype.ComponentEvent, gs *cptype.GlobalStateData) error {
+	bot.SDK = cputil.SDK(ctx)
+	projectid, _ := strconv.ParseUint(bot.SDK.InParams["projectId"].(string), 10, 64)
+	bot.ctx = ctx
+	var IssueDoshboardBatchSubmit cptype.OperationKey = "batchSubmit"
+	err := Transfer(c.Operations, &bot.Operations)
+	if err != nil {
+		return err
+	}
+	switch event.Operation {
+	case cptype.InitializeOperation:
+		bot.getOperations()
+		err := Transfer(bot.Operations, &c.Operations)
+		if err != nil {
+			return err
+		}
+		bot.getProps()
+		err = Transfer(bot.Props, &c.Props)
+		if err != nil {
+			return err
+		}
+		return nil
+	case cptype.RenderingOperation:
+		bot.State.Visible = false
+		operationKey := (*gs)["OperationKey"]
+		if operationKey == nil {
+			break
+		}
+		if operationKey == "delete" {
+			bot.State.Visible = true
+			selectRowKeys := (*gs)["SelectedRowKeys"].([]string)
+			bot.State.SelectedRowKeys = selectRowKeys
+			ops := Operation{}
+			err := Transfer(bot.Operations["onOk"], &ops)
+			if err != nil {
+				return err
+			}
+			ops.Meta.Type = "delete"
+			bot.Operations["onOk"] = ops
+			bot.Props.Content = bot.getContent(bot.SDK.I18n("Delete following items"), selectRowKeys)
+		}
+	case IssueDoshboardBatchSubmit:
+		bot.State.Visible = false
+		ops := Operation{}
+		err := Transfer(bot.Operations["onOk"], &ops)
+		if err != nil {
+			return err
+		}
+		if ops.Meta.Type == "delete" {
+			bot.ctx = ctx
+			cputil.MustObjJSONTransfer(&c.State, &bot.State)
+			bot.State.Visible = false
+			issueSvc := ctx.Value(types.IssueService).(pb.IssueCoreServiceServer)
+			bot.issueSvc = issueSvc
+			_, err = bot.DeleteItems(bot.State.SelectedRowKeys, projectid)
+			if err != nil {
+				return err
+			}
+			(*gs)["OperationKey"] = ""
+		}
+	}
+	return bot.SetComponent(c)
+}
+
+func (bot *BatchOperationTipModal) SetComponent(c *cptype.Component) error {
+	err := Transfer(bot.Props, &c.Props)
+	if err != nil {
+		return err
+	}
+	err = Transfer(bot.State, &c.State)
+	if err != nil {
+		return err
+	}
+	err = Transfer(bot.Operations, &c.Operations)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (bot *BatchOperationTipModal) getContent(tip string, nodeIDs []string) string {
+	content := fmt.Sprintf("%s\n", tip)
+	for _, id := range nodeIDs {
+		splits := strings.Split(id, "/")
+		content += fmt.Sprintf("%s\n", splits[0])
+	}
+	return content
+}
+
+func (bot *BatchOperationTipModal) DeleteItems(itemIDs []string, projectID uint64) (*pb.BatchDeleteIssueResponse, error) {
+	var req = &pb.BatchDeleteIssueRequest{}
+	req.Ids = itemIDs
+	req.ProjectID = projectID
+	k, err := bot.issueSvc.BatchDeleteIssues(bot.ctx, req)
+	if err != nil {
+		return nil, err
+	}
+	return k, err
+}
+
+// Transfer transfer a to b with json, kind of b must be pointer
+func Transfer(a, b interface{}) error {
```

Comment:
`Transfer` is necessary to add unit-test

Reply:
thank you for pointing out